rosco-web/src/main/groovy/com/netflix/spinnaker/rosco/endpoint/StatusEndpoint.groovy

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2016 Schibsted ASA.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.rosco.endpoint
+
+import com.netflix.spinnaker.rosco.api.BakeStatus
+import com.netflix.spinnaker.rosco.persistence.BakeStore
+import groovy.util.logging.Slf4j
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.actuate.endpoint.AbstractEndpoint
+import org.springframework.boot.context.properties.ConfigurationProperties
+import org.springframework.stereotype.Component
+
+
+@Component
+@Slf4j
+@ConfigurationProperties(prefix = "endpoints.status", ignoreUnknownFields = false)
+class StatusEndpoint extends AbstractEndpoint<Map<String, Object>> {
+
+  private final BakeStore bakeStore
+
+  private static enum Status {
+    RUNNING, IDLE
+  }
+
+  @Autowired
+  StatusEndpoint(BakeStore bakeStore) {
+    super("status", true, false)
+    this.bakeStore = bakeStore
+  }
+
+  @Override
+  public Map<String, Object> invoke() {
+    def bakes
+    def instanceStatus
+    def instanceIncompleteBakeIds = bakeStore.getThisInstanceIncompleteBakeIds()
+    if (instanceIncompleteBakeIds) {
+      instanceStatus = Status.RUNNING
+      bakes = bakesStatus(instanceIncompleteBakeIds)
+    } else {
+      instanceStatus = Status.IDLE
+      bakes = []
+    }
+    Collections.unmodifiableMap(["status": instanceStatus.name(), "bakes": bakes])
```

Comment:
Please use `return` operator. We've been inconsistent about this in the past, but prefer to use them when it's not a single-line closure or something similar.